### PR TITLE
[com_fields] Custom Fields Media Field if class is empty still class is shown as attribute

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for Issue #13400 

### Summary of Changes

Don't let the class tag alone without content

### Testing Instructions

Create an media field and don´t fill the image class field.

There is an empty class attribute shown in the Source code
image

Apply the patch and make sure there is no class tag anymore but if you set a class it comes back :smile: 

### Documentation Changes Required

none

### CODE REVIEW

**Comment by laoneo** [comment]
Can we put this one on hold till #13319 is merged. Then this one needs to be redone in the plugin itself. Thanks.

**Comment by zero-24** [comment]
@coolcat-creations @laoneo i have updated the PR after the changes in #13319 please review and test here.

**Comment by laoneo** [comment]
Please use `$fieldParams->get('image_class')` to get the image class. The inheritance from the plugin works.

**Comment by zero-24** [comment]
done @laoneo

**Comment by laoneo** [comment]
Just saw that getting the directory is also wrong, can you change it to `$fieldParams->get('directory', '/')` as well?

**Comment by zero-24** [comment]
fixed.

File: plugins/fields/media/tmpl/media.php [hunk 1]
```diff
@@ -25,7 +31,7 @@
 		continue;
 	}
 
-	$buffer .= '<img src="' . $path . '" class="' . $fieldParams->get('image_class') . '"/>';
+	$buffer .= '<img src="' . $path . '"' . $class . '"/>';
 }
```
Comment (laoneo) [inline plugins/fields/media/tmpl/media.php]: You have a not needed  " at the end of a tag and a space is needed

    $buffer .= '';

File: plugins/fields/imagelist/tmpl/imagelist.php [hunk 2]
```diff
@@ -25,8 +31,8 @@
 		continue;
 	}
 
-	$buffer .= '<img src="images/' . $field->fieldparams->get('directory', '/') . '/' . $path
-			. '" class="' . $field->fieldparams->get('image_class') . '"/>';
+	$buffer .= '<img src="images/' . $fieldParams->get('directory', '/')
+				. '/' . $path . '"' . $class . '"/>';
 }
```
Comment (laoneo) [inline plugins/fields/imagelist/tmpl/imagelist.php]: You have a not needed  " at the end of a tag and a space is needed

    . '/' . $path . '" ' . $class . '/>';

**Review by laoneo — CHANGES_REQUESTED** [review]
Produces invalid HTML

**Comment by franz-wohlkoenig** [comment]
I have tested this item :red_circle: unsuccessfully on f04480476a60a815f6f3b7c79cfd153028aa2af0<br /><br />Before Patch:
```
<dd class="field-entry " id="field-entry-8">
	<span class="field-label">Media: </span>
	<span class="field-value"><img src="/Joomla/images/joomla_black.png" class=""/></span>
</dd>
```
After Patch:
```
<dd class="field-entry " id="field-entry-8">
	<span class="field-label">Media: </span>
	<span class="field-value"><img src="/Joomla/images/joomla_black.png""/></span>
</dd>
</dl>
```
There ist a " to much at `joomla_black.png""`<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13462">issues.joomla.org/tracker/joomla-cms/13462</a>.</sub>

**Comment by zero-24** [comment]
@laoneo @franz-wohlkoenig the last commits should do it thanks!.

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 14c96a662802c96738b0659e0578b883d12c4e0f<br /><br />After PR empty Class is gone.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13462">issues.joomla.org/tracker/joomla-cms/13462</a>.</sub>

File: plugins/fields/imagelist/tmpl/imagelist.php [hunk 3]
```diff
@@ -25,8 +31,8 @@
 		continue;
 	}
 
-	$buffer .= '<img src="images/' . $field->fieldparams->get('directory', '/') . '/' . $path
-			. '" class="' . $field->fieldparams->get('image_class') . '"/>';
+	$buffer .= '<img src="images/' . $fieldParams->get('directory', '/')
+				. '/' . $path . '"' . $class . '/>';
 }
```
Comment (laoneo) [inline plugins/fields/imagelist/tmpl/imagelist.php]: There is still a space here missing.

File: plugins/fields/media/tmpl/media.php [hunk 4]
```diff
@@ -25,7 +31,7 @@
 		continue;
 	}
 
-	$buffer .= '<img src="' . $path . '" class="' . $fieldParams->get('image_class') . '"/>';
+	$buffer .= '<img src="' . $path . '"' . $class . '/>';
 }
```
Comment (laoneo) [inline plugins/fields/media/tmpl/media.php]: There is still a space here missing.

**Review by laoneo — CHANGES_REQUESTED** [review]

**Review by laoneo — APPROVED** [review]

File: plugins/fields/imagelist/tmpl/imagelist.php (hunk 3 above)
Comment (laoneo) [inline plugins/fields/imagelist/tmpl/imagelist.php]: Fixed with c1f1a4b

File: plugins/fields/media/tmpl/media.php (hunk 4 above)
Comment (laoneo) [inline plugins/fields/media/tmpl/media.php]: Fixed with 2a49a4f.

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on c1f1a4bdd8a444203422a9345e10ba5fad8b4947<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13462">issues.joomla.org/tracker/joomla-cms/13462</a>.</sub>

**Comment by laoneo** [comment]
@franz-wohlkoenig please test it with a class and without. For me it worked.

**Comment by laoneo** [comment]
@franz-wohlkoenig can you retest this one, that we can get it in?

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on fd069e6e9832d6f0c5938d0de9baed2ee7dc54af<br /><br />With and without a class (i.e. `span4 offset1`) class isn't shown in source code.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13462">issues.joomla.org/tracker/joomla-cms/13462</a>.</sub>

**Comment by laoneo** [comment]
Ready for commit then, thanks.

**Comment by jeckodevelopment** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13462">issues.joomla.org/tracker/joomla-cms/13462</a>.</sub>

**Pull request merged**